qemu/tests/blockdev_mirror_hotunplug.py

```diff
@@ -20,13 +22,23 @@ def _device_del(device):
 
         list(map(_device_del, self._source_images))
 
+    def record_number_of_device_before_delete(self):
+        """
+        record number of devices before delete
+        """
+        self._device_numbers = len(self.main_vm.monitor.query("block"))
+
     def wait_till_frontend_devices_deleted(self):
         """Wait till devices removed from output of query-block"""
         def _is_device_deleted(device):
-            for item in self.main_vm.monitor.query("block"):
-                if device in item["qdev"]:
-                    return False
-            return True
+            res = self.main_vm.monitor.query("block")
```

Comment:
I think,
  if device in item.get("qdev"):
        return False
is enough.

After check with ngu, the issue was caused as ppc can't get complete info in a immediate state for the deleted device. 
So if we can't get qdev, the device has been deleted in fact.
Or another way, if we can't get qdev, we can break this loop, enter next _is_device_deleted.

Reply:
> I think, if device in item.get("qdev"): return False is enough.
> 
> After check with ngu, the issue was caused as ppc can't get complete info in a immediate state for the deleted device. So if we can't get qdev, the device has been deleted in fact. 
>>> It's in an unstable state, maybe removing the device is still in progress.
Or another way, if we can't get qdev, we can break this loop, enter next _is_device_deleted.
>>> Safe way! Skip the unstable state, wait till the device is deleted. This is also OK from my side.

Reply:
hi @nickzhq , what do you think about Aihua's comments?  It looks a simple solution.